rfcs/0182-taskcluster-yml-remote-references.md

```diff
@@ -0,0 +1,49 @@
+# RFC 182 - Allow remote references to .taskcluster.yml files processed by Taskcluster-GitHub
+* Comments: [#182](https://github.com/taskcluster/taskcluster-rfcs/pull/182)
+* Proposed by: @bhearsum
+
+# Summary
+
+Allow `.taskcluster.yml` files in GitHub repositories to outsource most of their contents to a `.taskcluster.yml` stored elsewhere.
+
+## Motivation
+
+`.taskcluster.yml` files in GitHub repositories are responsible for creating tasks (or not) in response to various GitHub events. In some cases (eg: when a decision task is used to delegate most of this to a separate tool), these files end up looking almost entirely identical, and hundreds of lines of JSON-e ends up duplicated across many different repositories. This duplication is a significant maintenance burden, and easily results in unwanted differences.
+
+Allowing a `.taskcluster.yml` to reference a file stored elsewhere will allow for deduplication of the bulk of the content in these files, resolving the concerns noted above.
```

Comment:
Should we leave a note here saying that those external `.taskcluster.yml` should be a valid `version:1` tc yamls that can be validated with the same schema? Or those embedded files would be free-form yamls that only produce `tasks:`?

I think we could add a line here about such validity and also when implementation will go, github service would fail if embedded config has invalid schema

Reply:
@lotas - I pushed a change that makes the schema more clear - any thoughts?

Reply:
thanks for the updates @bhearsum. I feel like we have already something implementable here :)